packages/contracts-bedrock/test/scripts/L2Genesis.t.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Test } from "forge-std/Test.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+import { console2 as console } from "forge-std/console2.sol";
+
+import { Artifacts } from "scripts/Artifacts.s.sol";
+import { DeployConfig, DEV_ACCOUNT_FUND_AMT } from "scripts/DeployConfig.s.sol";
+import { Executables } from "scripts/Executables.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L2GenesisHelpers } from "scripts/libraries/L2GenesisHelpers.sol";
+
+/// @notice Reads a `genesis-l2.json` file, parses the `alloc`s, and runs assertions
+///         against each alloc depending on whether it's a precompile, predeploy proxy,
+///         or predeploy implementation.
+contract L2Genesis_Test is Test, Artifacts {
+    DeployConfig public constant cfg =
+        DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
+
+    string internal genesisPath;
+
+    struct StorageData {
+        bytes32 key;
+        bytes32 value;
+    }
+
+    /// @notice `balance` and `nonce` are being parsed as `bytes` even though their JSON representations are hex
+    /// strings.
+    ///         This is because Foundry has a limitation around parsing strings as numbers when using `vm.parseJson`,
+    ///         and because we're using `abi.decode` to convert the JSON string, we can't use coersion (i.e.
+    /// `vm.parseJsonUint`)
+    ///         to tell Foundry that the strings are numbers. So instead we treat them as `byte` strings and parse as
+    ///         `uint`s when needed. Additional context: https://github.com/foundry-rs/foundry/issues/3754
+    struct Alloc {
+        address addr;
+        bytes balance;
+        bytes code;
+        bytes nonce;
+        StorageData[] storageData;
+    }
+
+    /// @notice Checks if deploy config file exists at `deployConfigPath`, if it doesn't
+    ///         reverts with warning to run `make devnet-allocs` at monorepo root.
+    ///         Afterwards, checks if genesis file exists at `genesisPath`, and if not
+    ///         runs `L2Genesis.s.sol` script to generate it.
+    function setUp() public override {
+        super.setUp();
+        Artifacts.setUp();
+
+        string memory deployConfigPath = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        if (!vm.exists(deployConfigPath)) {
+            revert(
+                string.concat(
+                    "Did not find deploy config at: ",
+                    deployConfigPath,
```

Comment:
the deploy config is not generated by `make devnet-allocs`, we don't have a command for auto generating the deploy config

Reply:
`make devnet-allocs` [does](https://github.com/ethereum-optimism/optimism/blob/6120d22fc8382f92c990d2c81231fa8e981ed7dd/bedrock-devnet/devnet/__init__.py#L126-L137) trigger `devnetL1.json` to be generated, but recommending this cmd doesn't make sense it this script will be used for more than the devnet - will we remove the recommendation